content/devops-israel/drone-lambda/index.md

```diff
@@ -0,0 +1,99 @@
+---
+author: "devops-israel"
+date: 2017-10-22T08:00:35+01:00
+image: "omerxx/drone-lambda-plugin"
+repo: "omerxx/drone-lambda-plugin"
+tags: [ deploy, aws, lambda ]
+logo: amazon_lambda.svg
+title: AWS Lambda Deploy
+---
+
+The plugin utilizes AWS go-sdk to update an existing function's code; build your code, zip it with dependencies and upload it to S3. Then trigger the plugin for deploy.
+
+## Usage
+
+#### Execute from the working directory; 
+This will update `my-function` with a zip file under `S3://some-bucket/lambda/lambda-project-1.zip`:
+```bash
+docker run --rm \
+  -e PLUGIN_FUNCTION_NAME=my-function \
+  -e PLUGIN_S3_BUCKET=some-bucket \
+  -e PLUGIN_FILE_NAME=lambda-project-1.zip \
+  omerxx/drone-lambda-plugin
+```
+
+#### Example:
+
+```yaml
+pipeline:
+  deploy-lambda:
+    image: omerxx/drone-lambda-plugin
+    pull: true
+    function_name: my-function
+    s3_bucket: some-bucket
+    file_name: lambda-dir/lambda-project-${DRONE_BUILD_NUMBER}.zip
+```
+
+#### Example of a complete Lambda project's pipeline:
+
+```yaml
+pipeline:
+  build:
+    image: python:2.7-alpine
+    commands:
+      - apk update && apk add zip
+      - pip install -r requirements.txt -t .
+      - zip -r -9 lambda-project-${DRONE_BUILD_NUMBER}.zip *
+
+  s3-publish:
+    image: plugins/s3
+    acl: private
+    region: us-east-1
+    bucket: some-bucket
+    target: lambda-dir
+    source: lambda-project-${DRONE_BUILD_NUMBER}.zip
+
+  deploy-lambda:
+    image: omerxx/drone-lambda-plugin
+    pull: true
+    function_name: my-function
+    s3_bucket: some-bucket
+    file_name: lambda-dir/revenue-report-${DRONE_BUILD_NUMBER}.zip
+
+  notify-slack-releases:
+    image: plugins/slack
+    channel: product-releases
+    webhook: https://hooks.slack.com/services/ABCD/XYZ
+    username: Drone-CI
+```
+
+## Optional secrets
```

Comment:
Please format secrets like this: http://plugins.drone.io/drone-plugins/drone-ecr/#secret-reference

Note: not everyone using this will be running Drone on AWS, so they may not have access to the default instace IAM roles.

Reply:
Done.

Thanks for the comment, I assumed that Lambda users would be using aws for everything. Good note. I added a small comment to make this more clear.